glean-core/android/src/main/java/mozilla/telemetry/glean/net/PingUploader.kt

```diff
@@ -22,4 +35,82 @@ internal interface PingUploader {
         dateFormat.timeZone = TimeZone.getTimeZone("GMT")
         return dateFormat.format(calendar.time)
     }
+
+    /**
+     * Function to deserialize and process all serialized ping files.  This function will ignore
+     * files that don't match the UUID regex and just delete them to prevent files from polluting
+     * the ping storage directory.
+     *
+     * @param processingCallback Callback function to do the actual process action on the ping.
+     *                           Typically this will be the [HttpPingUploader.upload] function.
+     * @return Boolean representing the success of the upload task. This may be the value bubbled up
+     *         from the callback, or if there was an error reading the files.
+     */
+    fun process(processingCallback: (String, String, Configuration) -> Boolean): Boolean {
```

Comment:
Since this is now living in `PingUploader.kt`, there's no need to pass in a callback. You should just be able to call `this.upload`